### ISSUE TITLE
Lab 8 Test Case 6.1

### ISSUE THREAD

**Issue by teekayowh** [nus-cs2030/2122-s2#267]
<img width="309" alt="cs2030" src="https://user-images.githubusercontent.com/70503149/161951798-36471b18-189c-4aa2-abb9-6119b130afb9.png">
![image](https://user-images.githubusercontent.com/70503149/161951937-e0eddbfa-81c6-402d-b6eb-1331c80d2a02.png)

 Hi, is anyone facing the same issue? This is my code and I am not sure why the list is updating...

**Comment by TohDeKai** [comment]
When there's no previous room, make sure to return the current room!

**Comment by alwinngjw** [comment]
Hi, I think this error comes from your code not returning the current room when there is no previous room.

Maybe you can modify your code to check for the existence of a previous room if it's there returns it. Else return the current room 

That may solve your error

Hope this helps.

**Comment by junbin-tan** [comment]
You may want to use .orElse(this) to solve your problem.

Hope it helps

**Comment by ReganKvioke** [comment]
I agree with @alwinngjw, maybe you want to check whether the previous state of the room remains the same (unless you take the sword). This may be because the previous state of the room is empty and there is a small typo or error there. Hopefully this helps.

**Comment by whyyouj** [comment]
Hello! Under your back method, if the previous room is the current room (which means current room is the first room), then you can return this (same instance) instead of making a new Room with the same items. In my case, I replaced `prevRoom` with `this` and it worked! Hope this helps you!
